api/src/main/java/jakarta/data/metamodel/restrict/CompositeRestrictionRecord.java

```diff
@@ -43,4 +43,31 @@ record CompositeRestrictionRecord<T>(
     public CompositeRestriction<T> negate() {
         return new CompositeRestrictionRecord<>(type, restrictions, !isNegated);
     }
+
+    /**
+     * Textual representation of a composite restriction.
+     * For example,
+     * <pre>ALL (price LESS_THAN 50.0, name LIKE "%Jakarta EE%")</pre>
+     *
+     * @return textual representation of a composite restriction.
+     */
+    @Override
+    public String toString() {
+        StringBuilder s = new StringBuilder(200);
```

Comment:
Why this constructor have size and it does not happen with `BasicRestrictRecord`

I also would recommend magic number, generate a constant and explain about this `200`

Reply:
Initially, I didn't supply a size, but the static analysis code checking tools that are built in to the build told me to add it. Apparently they didn't notice BasicRestrictRecord.  I updated all 3 methods with more precise estimates and made this one use a constant as you suggested in the latest commit.